fedimint-api/src/config.rs

```diff
@@ -19,41 +21,179 @@ use tbs::hash::hash_bytes_to_curve;
 use tbs::poly::Poly;
 use tbs::serde_impl;
 use tbs::Scalar;
+use url::Url;
 
 use crate::cancellable::Cancellable;
-use crate::net::peers::AnyPeerConnections;
-use crate::task::TaskGroup;
+use crate::multiplexed::{ModuleIdRef, ModuleMultiplexer};
+use crate::Amount;
 use crate::PeerId;
 
-/// Part of a config that needs to be generated to bootstrap a new federation.
-#[async_trait(? Send)]
-pub trait GenerateConfig: Sized {
-    type Params: ?Sized;
-    type ClientConfig;
-    type ConfigMessage;
-    type ConfigError;
-
-    /// Function that generates the config of all peers locally. This is only meant to be used for
-    /// testing as the generating machine would be a single point of failure/compromise.
-    fn trusted_dealer_gen(
-        peers: &[PeerId],
-        params: &Self::Params,
-        rng: impl RngCore + CryptoRng,
-    ) -> (BTreeMap<PeerId, Self>, Self::ClientConfig);
-
-    fn to_client_config(&self) -> Self::ClientConfig;
-
-    /// Asserts that the public keys in the config are and panics otherwise (no way to recover)
-    fn validate_config(&self, identity: &PeerId);
-
-    async fn distributed_gen(
-        connections: &mut AnyPeerConnections<Self::ConfigMessage>,
-        our_id: &PeerId,
-        peers: &[PeerId],
-        params: &Self::Params,
-        rng: impl RngCore + CryptoRng,
-        task_group: &mut TaskGroup,
-    ) -> Result<Cancellable<(Self, Self::ClientConfig)>, Self::ConfigError>;
+#[derive(Debug, Clone, Eq, PartialEq, Hash, Serialize, Deserialize)]
+pub struct Node {
+    pub url: Url,
+    pub name: String,
+}
+
+/// Total client config
+///
+/// This includes global settings and client-side module configs.
+#[derive(Debug, Clone, Eq, PartialEq, Serialize, Deserialize)]
+pub struct ClientConfig {
+    pub federation_name: String,
+    pub nodes: Vec<Node>,
+    pub modules: BTreeMap<String, ClientModuleConfig>,
+}
+
+impl ClientConfig {
+    pub fn get_module<T: DeserializeOwned>(&self, module: &str) -> anyhow::Result<T> {
+        if let Some(client_cfg) = self.modules.get(module) {
+            Ok(serde_json::from_value(client_cfg.0.clone())?)
+        } else {
+            Err(format_err!("Client config for {module} module not found"))
+        }
+    }
+}
+
+/// Global Fedimint configuration generation settings passed to modules
+///
+/// This includes typed module settings for know modules for simplicity,
+/// and better UX, while the non-standard modules have to use a type-erased
+/// config.
+///
+/// Candidate for re-designing when the modularization effort is
+/// complete.
+pub struct ModuleConfigGenParams {
+    pub mint_amounts: Vec<Amount>,
+    pub bitcoin_rpc: BitcoindRpcCfg,
+
+    /// extra options for extra settings and modules
+    pub other: BTreeMap<String, serde_json::Value>,
+}
```

Comment:
~Not sure this will fly with modularization, you'll have to remove these concrete fields imo if we don't want the mint+wallet module to become some special case.~

Now I see what you did there: you just give the entire struct to every module and let it figure out what to read from it. An approach similar to how config is handled in this PR would be better imo, but not a blocker.